ipatests/test_integration/test_forced_client_reenrollment.py

```diff
@@ -43,6 +43,17 @@ class TestForcedClientReenrollment(IntegrationTest):
     def install(cls, mh):
         super(TestForcedClientReenrollment, cls).install(mh)
         tasks.install_master(cls.master)
+        # Get zone the client belongs to.
+        cls.host_zone = tasks.get_host_zone(cls.master, cls.clients[0])
+        # In cases where client is managed by upstream DNS server we overlap
+        # its zone so we can save DNS records (e.g. SSHFP) for comparison.
+        if cls.host_zone:
+            servers = [cls.master] + cls.replicas
+            tasks.add_dns_zone(cls.master, cls.host_zone,
+                               skip_overlap_check=True,
+                               dynamic_update=True,
```

Comment:
I'm not sure why this test is here, in force_client_reenrollment vs an original master installation. It seems to that domain overlap is best tested in either a new server install or ipa-dns-install.

Reply:
This is whole purpose of the patch. The test was actually never passing in our LAB as 'nsupdate' from client was trying to update the upstream DNS server. In that case we overlap the upstream zone so we can get the records updated. If there is no overlap we continue with master domain (e.g. ipa.test).
Previously this test was running in isolated environment in a spare workstation, however we decided to make the necessary fixes to make it running. This was also consulted with @pvoborni 
After this patch the test is passing and looks stable.